lib/HTFeed/StorageZipAudit/ObjectStore.pm

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/perl
+package HTFeed::StorageZipAudit::ObjectStore;
+
+use strict;
+use warnings;
+use Carp;
+use Log::Log4perl qw(get_logger);
+use File::Temp;
+
+use HTFeed::Config qw(get_config);
+use HTFeed::DBTools qw(get_dbh);
+use HTFeed::Volume;
+
+use base qw(HTFeed::StorageZipAudit);
+
+sub storage_object_path {
+  my $self = shift;
+  my $obj  = shift;
+
+  unless ($obj->{tmpdir}) {
+    $obj->{tmpdir} = File::Temp->newdir();
+  }
+  return $obj->{tmpdir};
+}
+
+sub all_objects {
```

Comment:
I like the approach here of restoring restorable objects and then essentially making this act like any other storage.